### DISCUSSION TITLE
Any particular reason to go straight to Go 1.18 ?

### DISCUSSION THREAD

**Discussion by alvarowolfx** [FerretDB/FerretDB#152] [General]
I'm starting to contribute to the project ( as per PR #151 ) and I was doing it using Go 1.17 running locally. I did so because it was a bit hard for me is using Go 1.18 locally. I use GVM to manage multiple versions of Go locally and seems like it doesn't support installing from the tip of Go. 

Also, recently all references to `interface{}` were changed to `any`, which made the project not backward compatible, even that is still not using any feature specific from Go 1.18 ( like generics ) as I still can run it here with Go 1.17 ( I just replaced all `any` types back to `interface{}`.

Not trying to enforce something here, I just wanted to understand the reasons behind jumping that fast to 1.18 and also making the project no backward compatible. I'll try to install 1.18 here, but would be nice to have more contributors if we can use more versions of Go.

Awesome project btw and thanks for sharing it.

**Comment by AlekSi** [comment]
> even that is still not using any feature specific from Go 1.18 ( like generics )

Generics are indeed used, for example, by that [trivial function](https://github.com/AlekSi/pointer/blob/039be97b958cc3cbeb2329617d2e493bd2331fac/generic.go#L6-L9) that is used in tests. More importantly, protocol decoding tests rely heavily on [fuzzing](https://go.dev/blog/fuzz-beta).

**Comment by alvarowolfx** [comment]
Now Go1.18 beta 1 is available through GVM, so it's going to be easier to use go with generics and contribute to the project 🎉
